src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/FSharp/FSharpProjectSelector.cs

```diff
@@ -38,5 +41,22 @@ internal static void GetProjectFactoryGuid(XDocument doc, out Guid guidProjectFa
 
             guidProjectFactory = ProjectType.LegacyFSharpGuid;
         }
+
+        public static async Task<IDisposable> RegisterAsync(ManagedProjectSystemPackage package)
```

Comment:
This doesn't need a package, just IAsyncServiceProvider.

Reply:
In the latest update it was necessary to pass the package so it could call back for `RegisterProjectFactory`.